server/controllers/errors/index.js

```diff
@@ -1,9 +1,19 @@
+const createError = require('http-errors');
+
 exports.notFound = (req, res) => {
-  res.send('not found');
+  res.status(404).send(createError(404, 'Page Not Found ...!'));
 };
 
-
 exports.serverError = (err, req, res, next) => {
-  console.log(err);
-  res.send('Error 500');
+  switch (err.code) {
+    case 401:
+      res.status(400).send(createError(400, err.msg));
```

Comment:
@Fatmasiam I think you should still send back a `401` status here, and not a `400`, so that the client knows that there was an authentication problem (then the client could choose to redirect the user to the login page, for example).  With a 400, the client would just think that there was some other generic server error.

Reply:
@tomduggan85 
Thanks for taking the time to do code review.
I think here the code error should be 400
```js
case 400:
```
we agreed to send the validation errors with this code. I am not sure what is the proper status for it :confused:

Reply:
@ashatat You're welcome.  I think 400 is good for validation errors, and also keep 401 for auth errors:
```
case 400:
      res.status(400).send(createError(400, err.msg)); /* For validation errors or other cases where client made a bad request */
case 401:
      res.status(401).send(createError(401, 'authentication error')); /* Auth failed */
case 403:
      res.status(403).send(createError(403, 'forbidden')); /* Forbidden: authentication was successful - backend thinks that the user is who they say they are - but that user is not allowed to access that endpoint, for example a "normal" user trying to do an "admin" action */
case 500:
     ....
```